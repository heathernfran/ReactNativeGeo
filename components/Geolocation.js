import React from 'react';
import { AppRegistry, Platform, StyleSheet, Text, View } from 'react-native';
import { Constants, Location, Permissions } from 'expo';

export default class Geolocation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      lat: '0',
      lng: '0',
      errorMessage: null,
    };
  }

  componentWillMount() {
    if (Platform.OS === 'android' && !Constants.isDevice) {
      this.setState({
        errorMessage: 'Does not work on Android emulator or Sketch'
      });
    } else {
      this.getLocation();
    }
  }

  getLocation = async () => {
    // Check that permission is granted for device
    let { status } = await Permissions.askAsync(Permissions.LOCATION);
    if (status !== 'granted') {
      this.setState({
        errorMessage: 'Permission to access location is not allowed'
      });
    }

    // Coordinates for latitude and longitude are set
    let currentPosition = await Location.getCurrentPositionAsync({});
    this.setState({
      lat: currentPosition.coords.latitude,
      lng: currentPosition.coords.longitude,
    });
  }

  render() {
    return (
      <View>
        <Text>Current position</Text>
        <Text>{`latitude: ${JSON.stringify(this.state.lat)}, longitude: ${JSON.stringify(this.state.lng)}`}</Text>
      </View>
    );
  }
}
