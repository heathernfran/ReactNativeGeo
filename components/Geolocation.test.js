import React from 'react';
import Geolocation from './Geolocation';

import renderer from 'react-test-renderer';

// beforeEach(() => {
  // let component = renderer.create(<Geolocation />);
  let component = shallow(<Geolocation />);
// });

it('renders the Geolocation component', () => {
  const snapshot = renderer.create(<Geolocation />).toJSON();
  expect(snapshot).toMatchSnapshot();
});

it.only('renders text', () => {
  const text = component.find('Text').children().text();
  expect(text).toMatch(/Current/);
});

// it('has permission to access location', () => {
//   expect(errorMessage).toBeNull();
// });

// it('gets latitude and longitude coordinates', () => {
//   const getLocation = jest.fn();
//   expect(getLocation.mock.instances.lat).not.toBe(0);
//   // expect(lng).not.toBe(0);
// });
