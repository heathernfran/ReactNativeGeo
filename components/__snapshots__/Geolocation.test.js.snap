// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Geolocation component 1`] = `[Function]`;
